rktl_sim/nodes/simulator

```diff
@@ -358,5 +449,244 @@ class Simulator(object):
+            raise self.InvalidRospyParam(path)
+        else:
+            type_rospy = type(rospy.get_param(path))
+
+            if type_rospy == dict:
+                min_param = rospy.get_param(f'${path}/min')
+                max_param = rospy.get_param(f'${path}/max')
+
+                if not max_param:
+                    raise self.InvalidRospyParam(f'${path}/max')
+                if not min_param:
+                    raise self.InvalidRospyParam(f'${path}/min')
+                # accounting for moments in yaml file
+                if min_param > max_param:
+                    return random.uniform(max_param, min_param)
+                else:
+                    return random.uniform(min_param, max_param)
+
+            elif type_rospy == float or type_rospy == int:
+                return rospy.get_param(path)
+
+    def get_sim_param(self, path):
+        """
+        create a dictionary from the path variable
+
+        process:
+        check if the initial path exist in rospy. if not, throw error
+        check if its a dict, if so, check if have min and max, if so, generate random param
+        if not, return a value if its a float, integer or list
+
+        if not have min and max, for each key of the dictionary, recurse down the tree by calling the same function
+
+        @param path: the path to the parameter in the yaml file
+        @return: either the variable or the complete dictionary, depending on the path taken during recurssion
+        """
+        rospy_param = rospy.get_param(path)
+        if not rospy_param:
+            raise self.InvalidRospyParam(path)
+        else:
+            type_rospy = type(rospy_param)
+            if type_rospy == dict:
+                # check here if there are min and max values specified
+                # if so, return the random values
+                min_param = rospy.get_param(f'${path}/min')
+                max_param = rospy.get_param(f'${path}/max')
+                if not max_param and min_param:
+                    raise self.InvalidRospyParam(f'${path}/max')
+                elif not min_param and max_param:
+                    raise self.InvalidRospyParam(f'${path}/min')
+                elif max_param and min_param:
+                    if min_param > max_param:
+                        return random.uniform(max_param, min_param)
+                    else:
+                        return random.uniform(min_param, max_param)
+
+                # bc not have max and min, this means that we have more dicts and vals to loop over
+                for key in rospy_param:
+                    #recurse down the path, using the key as the continuation
+                    rospy_param[key] = self.get_sim_param(path + f'$/{key}')
+                return rospy_param
+
+            elif type_rospy == float or type_rospy == int or type_rospy == list:
+                return rospy.get_param(path)
+
+        raise self.InvalidRospyParam(path)
```

Comment:
We don't want to raise an exception in a ROS node. Instead we should handle error, log and shutdown cleanly.

Reply:
I made the InvalidRospyParam exception to be in charge of the error. the implementation is below for this specific exception: 
`   class InvalidRospyParam(Exception):
"""exception when using invalid arguement"""
print(Exception)
rospy.logfatal()
`
this is located in the simulator node

Reply:
Does this work, or are you looking over for something else?